### ISSUE TITLE
Rule Proposal: `no-v-else`

### ISSUE THREAD

**Issue by fisker** [vuejs/eslint-plugin-vue#1489]
**Please describe what the rule should do:**

Forbid use `v-else-if` and `v-else`, I don't like use `v-else` because I can't simply remove the component with `v-if`.

**What category should the rule belong to?**

[ ] Enforces code style (layout)
[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[x] Other (please specify:) Improve editability

**Provide 2-3 code examples that this rule should warn about:**

```vue
<!-- Bad -->
<template>
  <div>
    <div v-if="foo"></div>
    <div v-else></div>
  </div>
</template>


<!-- Good -->
<template>
  <div>
    <div v-if="foo"></div>
    <div v-if="!foo"></div>
  </div>
</template>
```

```vue
<!-- Bad -->
<template>
  <div>
    <div v-if="foo === 1"></div>
    <div v-else-if="foo === 2"></div>
    <div v-else></div>
  </div>
</template>

<!-- Good -->
<template>
  <div>
    <div v-if="foo === 1"></div>
    <div v-if="foo === 2"></div>
    <div v-if="foo !== 1 && foo !== 2"></div>
  </div>
</template>
```

**Additional context**

N/A

**Comment by ota-meshi** [comment]
Thank you for suggesting the rules.

> I don't like use v-else because I can't simply remove the component with v-if.

Could you explain the details on this? We may need to handle it depending on the v-else-if condition.

**Comment by ota-meshi** [comment]
Thank you for explaining.

It seems that Vue2 doesn't allow you to put multiple `v-if` on the root element, so the rule may need some options or handles.

https://template-explorer.vuejs.org/#%3Cdiv%20id%3D%22app%22%20v-if%3D%22a%22%3E%7B%7B%20msg%20%7D%7D%3C%2Fdiv%3E%0A%3Cdiv%20id%3D%22app%22%20v-if%3D%22!a%22%3E%7B%7B%20msg%20%7D%7D%3C%2Fdiv%3E

**Comment by fisker** [comment]
Nice catch!

**Comment by fisker** [comment]
I wasn't expecting auto-fix for it, may just simply forbid `VDirectiveKey[name="else-if"]` and `VDirectiveKey[name="else"]`

**Comment by ota-meshi** [comment]
> But vue2 requires one single root too

If you use v-if and v-else-if with Vue2, there is no problem even if there are no elements.

https://template-explorer.vuejs.org/#%3Cdiv%20id%3D%22app%22%20v-if%3D%22a%22%3E%7B%7B%20msg%20%7D%7D%3C%2Fdiv%3E%0A%3Cdiv%20id%3D%22app%22%20v-else-if%3D%22b%22%3E%7B%7B%20msg%20%7D%7D%3C%2Fdiv%3E

**Comment by fisker** [comment]
Close this as I'm going to implement myself.
